File: EPI/7.8/swift/7.8_test.swift
import Darwin

let tests: [(index: Int, result: String)] = [
    (0, "1"),
    (1, "11"),
    (2, "21"),
    (3, "1211"),
    (4, "111221"),
    (5, "312211"),
    (6, "13112221"),
    (7, "1113213211"),
]

for test in tests {
    var actual = lookAndSay(index: test.index)
    guard actual == test.result else {
        print("For index \(test.index), expected result to be '\(test.result)', " +
            "but was '\(actual)'")
        exit(1)
    }
}

File: EPI/Strings/The look-and-say problem/typescript/7.8_test.ts
import lookAndSay from './7.8';

declare function require(name: string): any;
const assert = require('assert');

const tests: {n: number, result: string}[] = [
    {n: 0, result: '1'},
    {n: 1, result: '11'},
    {n: 2, result: '21'},
    {n: 3, result: '1211'},
    {n: 4, result: '111221'},
    {n: 5, result: '312211'},
    {n: 6, result: '13112221'},
    {n: 7, result: '1113213211'},
];

for (const test of tests) {
    const actual = lookAndSay(test.n)
    assert.strictEqual(actual, test.result,
        `For n ${test.n}, expected result to be ${test.result}, but was ${actual}`);
}
